modules/ROOT/pages/build-application-from-api.adoc

```diff
@@ -0,0 +1,198 @@
+
+.REST API Main Flow.
+image::mruntime-soap-main-flow.png[SOAP API Main Flow]
+
+=== Backend Flows
+APIkit for REST generates a backend flow for each resource-action pairing in a RAML file. +
+APIkit for SOAP generates a backend flow for each operation in a WSDL. +
+In the following example, the RAML interface receives the GET request users. The backend flow accesses a database to look up users by id.
+
+.REST API Backend Flows.
+image::mruntime-rest-backend-flows.png[REST API Backend Flows]
+
+=== Simulating Calls
+For REST APIs, use the API Console to simulate calls to the API. +
+For SOAP APIs, use a SOAP testing tool such as SoapUI, to simulate calls to the API.
+
+== Start APIkit Project and Generate Mule Flows
+When you create an APIkit project in Studio, you have the option to whether import or not an API definition file and and when you do so, the APIkit scaffolding technique generates different flows for the API. There are three ways to start an APIkit project:
+
+=== Start a New APIkit Project with an API Definition File
+Create a project importing a RAML or WSDL file, or reference the file URL:
+
+. Select *File > New > Mule Project*.
+. In *Project Name*, enter a name for the project, for example *myproject*. Select or accept the default EE Runtime, for example *Mule Server 4.2.0 EE*.
+. In *API Implementation*, check *Specify API Definition File Location or URL*.
+. In *Location*, do one of the following:
+* Select *Design Center* if you created an `api.raml` file in Design Center. Login to Anypoint Platform if necessary, and select `api.raml`.
+* Alternatively, if you didn’t create the RAML in Design Center, select *Browse Files*. Browse to and select the RAML or WSDL file that you created in a text editor.
+* For WSDL files, select a service and port from the drop-down menus or accept the defaults.
+
+[start=5]
+. Accept the Location default options, and click *Finish*.
+
+.New Mule Project window selecting API specification .
+image::mruntime-start-apikit-project-1.png[New Mule Project with API Specification selected]
+
+The APIkit scaffolding technique generates:
+
+* An archetype that contains basic configuration files plus a Mule XML configuration file.
+* An implementation of the API that contains:
+** For RAML-based APIs:
+*** The RAML file appears under the `src/main/resources/api` folder.
+*** A main flow with an HTTP Listener.
+*** APIkit Router component.
+*** Error handlers.
+*** RAML dependencies, push or pull changes from one location, such as Studio, to the other, such as Design Center.
+*** Backend flows containing REST resource-action mappings.
+** For WSDL-based APIs:
+*** The WSDL file appears under the `src/main/resources/api` folder.
+*** A SOAP Router.
+*** A SOAP fault response template.
+*** Backend flows containing SOAP message templates.
+
+The below graphics explain the APIkit scaffolding technique:
+
+.RAML Scaffolding.
+image::mruntime-raml-scaffolding.png[RAML Scaffolding]
```

Comment:
all this is going to chance since studio 7.4.0 GA on decembeer, if you can track this would be great 

cc @martincousido

Reply:
Yes, I'll track it. Thanks for the heads up.